### PR TITLE
Added fill-up with pre-existing API

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

File: cl/phase1/core/checkpoint.go [hunk 1]
```diff
@@ -48,3 +51,51 @@ func RetrieveBeaconState(ctx context.Context, beaconConfig *clparams.BeaconChain
...
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = r.Body.Close()
+	}()
+	if r.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("checkpoint sync failed, bad status code %d", r.StatusCode)
+	}
+	marshaled, err := io.ReadAll(r.Body)
```
Comment (elee1766) [inline cl/phase1/core/checkpoint.go:76]: read content-length header first before unmarshaling. (early exit)

File: cmd/capcli/cli.go [hunk 2]
```diff
@@ -437,6 +441,116 @@ func (c *Chain) Run(ctx *Context) error {
...
+	// Let's fetch the head first
+	currentBlock, err := core.RetrieveBlock(ctx, beaconConfig, genesisConfig, fmt.Sprintf("%s/head", baseUri), nil)
+	if err != nil {
+		return err
+	}
+	currentRoot, err := currentBlock.Block.HashSSZ()
+	if err != nil {
+		return err
+	}
+	tx, err := db.BeginRw(ctx)
```
Comment (elee1766) [inline cmd/capcli/cli.go:479]: tx not closed

File: cmd/capcli/cli.go [hunk 3]
```diff
@@ -437,6 +441,116 @@ func (c *Chain) Run(ctx *Context) error {
...
+	if err := tx.Commit(); err != nil {
+		return err
+	}
+	previousLogBlock := currentBlock.Block.Slot
+
+	logInterval := time.NewTicker(30 * time.Second)
+	defer logInterval.Stop()
+
+	for {
+		tx, err := db.BeginRw(ctx)
```
Comment (elee1766) [inline cmd/capcli/cli.go:497]: tx not closed :(

**Review by elee1766 — APPROVED** [review]
change notes then fine

File: cl/phase1/core/checkpoint.go (hunk 1 above)
Comment (Giulio2002) [reply cl/phase1/core/checkpoint.go:76]: I dont think it matters as it is a cli for development

**Pull request merged**